data_structures/heap/heap_generic.py

```diff
@@ -1,35 +1,38 @@
+from collections.abc import Callable
+
+
 class Heap:
     """
     A generic Heap class, can be used as min or max by passing the key function
     accordingly.
     """
 
-    def __init__(self, key=None):
+    def __init__(self, key: Callable | None = None) -> None:
         # Stores actual heap items.
-        self.arr = list()
+        self.arr: list = list()
         # Stores indexes of each item for supporting updates and deletion.
-        self.pos_map = {}
+        self.pos_map: dict = {}
```

Comment:
These are really not needed.  Both Python and mypy are able to figure this out without hinting.

Reply:
i got an error in pipeline without adding this: https://results.pre-commit.ci/run/github/63476337/1665575464.hZt5fgARRZacJLt5kikIdg @cclauss

Reply:
@cclauss if you are satisfied, can you please approve? i cannot open more PRs in the project since i have mutliple open - https://github.com/TheAlgorithms/Python/pull/7043#issuecomment-1276023849